src/devices/cpu/drcbec.cpp

```diff
@@ -972,14 +988,27 @@ int drcbe_c::execute(code_handle &entry)
 
 			case MAKE_OPCODE_SHORT(OP_MULS, 4, 1):
 				temp64 = mul_32x32(PARAM2, PARAM3);
-				temp32 = (int32_t)temp64;
-				flags = FLAGS32_NZ(temp32);
+				flags = FLAGS64_NZ(temp64);
 				PARAM1 = temp64 >> 32;
 				PARAM0 = (uint32_t)temp64;
 				if (temp64 != (int32_t)temp64)
 					flags |= FLAG_V;
 				break;
 
+			case MAKE_OPCODE_SHORT(OP_MULSLW, 4, 0):      // MULSLW   dst,src1,src2[,f]
+				temp64 = mul_32x32(PARAM1, PARAM2);
+				PARAM0 = (int32_t)temp64;
+				break;
+
+			case MAKE_OPCODE_SHORT(OP_MULSLW, 4, 1):
+				temp64 = mul_32x32(PARAM1, PARAM2);
+				temp32 = (int32_t)temp64;
+				flags = FLAGS32_NZ(temp32);
+				PARAM0 = temp32;
+				if (temp64 != (int32_t)temp64)
+					flags |= FLAG_V;
+				break;
```

Comment:
These are playing fast and loose with undefined corners of integer conversion.  Will it work if you do it like this?

```c++
			case MAKE_OPCODE_SHORT(OP_MULSLW, 4, 0):      // MULSLW   dst,src1,src2[,f]
				temp64 = mul_32x32(PARAM1, PARAM2);
				PARAM0 = (uint32_t)temp64;
				break;

			case MAKE_OPCODE_SHORT(OP_MULSLW, 4, 1):
				temp64 = mul_32x32(PARAM1, PARAM2);
				temp32 = (uint32_t)temp64;
				flags = FLAGS32_NZ(temp32);
				PARAM0 = temp32;
				if ((int64_t)temp64 != (int32_t)temp32)
					flags |= FLAG_V;
				break;
```

Reply:
From a quick test, yeah this works fine. All of my test cases pass with the change. I'll change it.